lib/pages/goal_page.dart

```diff
@@ -3,102 +3,485 @@ import 'package:linear/model/goal.dart';
+       });
+       setState(() {
+         _goals = _allGoals.toList();
+       });
+     getOpen(_goals);
+     isSorting = true;
+     sortingBy = "Newest";
+     filteringBy= "Open";
+     setState(() {
+         _isLoading = false;
+         _isErrorFetchingGoals = false;
+       });
+ 
+     } else {
+       setState(() {
+          _allGoals = [];
+         _isLoading = false;
+         _isErrorFetchingGoals = true;
+       });
+     }
+   });
+ }
 
-  fetchGoals() {
-    final Future<Map<String, dynamic>> successfulMessage = api.getGoalsForGoalPage(context);
-    successfulMessage.then((response) {
-      if (response['status'] == true) {
-        setState(() {
-          _goals = response['goals'];
-          _isLoading = false;
-          _isErrorFetchingGoals = false;
-        });
-      } else {
-        setState(() {
-          _goals = [];
-          _isLoading = false;
-          _isErrorFetchingGoals = true;
-        });
-      }
-    });
-  }
+ getOpen(List getGoals){
+  
+   setState(() {
+    _goals = _allGoals.toList();
+  });
+  setZero = false;
+   Goal checkGoal = Goal.fromJson(_goals[0]);
+      for(int i = 0; i < _goals.length; i++){
+       checkGoal = Goal.fromJson(_goals[i]);
+       if((checkGoal.isFinished) == true){
+            setState(() {
+             _goals.removeAt(i);
+            i--;
+           });
+       }
+     }
+ }
 
-  @override
-  Widget build(BuildContext context) {
-      return Scaffold(
-        appBar: AppBar(
-          title: const Text("Goals"),
-          automaticallyImplyLeading: false,
-        ),
-        body: Center(
-            child: _isLoading?
-              const CircularProgressIndicator()
-              : _isErrorFetchingGoals?
-                const Text(
-                  "We ran into an error trying to obtain the goals. \nPlease try again later.",
-                  textAlign: TextAlign.center,
-                )
-                : buildGoalsList(),
-              
+ getOverdue(List getGoals){
+  setState(() {
+    _goals = _allGoals.toList();
+  });
+  Goal checkGoal = Goal.fromJson(_goals[0]);
+      for(int i = 0; i < _goals.length; i++){
+       checkGoal = Goal.fromJson(_goals[i]);
+       if((checkGoal.isFinished) == true || (checkGoal.completedCheckIns < checkGoal.checkInGoal)){
+            setState(() {
+             _goals.removeAt(i);
+            i--;
+           });
+       }
+     }
+ }
+
+ getFinished(List getGoals){
+  setState(() {
+    _goals = _allGoals.toList();
+  });
+   Goal checkGoal = Goal.fromJson(_goals[0]);
+      for(int i = 0; i < _goals.length; i++){
+       checkGoal = Goal.fromJson(_goals[i]);
+       if((checkGoal.isFinished) != true){
+            setState(() {
+             _goals.removeAt(i);
+             i--;
+           });
+       }
+     }
+ }
+
+findSearchResults(){
+  String community = "";
+   if(searchInput.text.isEmpty) {
+      return;
+    }else{
+      community = searchInput.text.toString();
```

Comment:
probably change to all lowercase as well assuming the input allows uppercase and causes issues, if not then no worries

Reply:
changed to lowercase